lib/shallow_attributes/class_methods.rb

```diff
@@ -66,8 +74,7 @@ def attributes
     def attribute(name, type, options = {})
       options[:default] ||= [] if type == Array
 
-      @default_values ||= {}
-      @default_values[name] = options.delete(:default)
+      default_values[name] = options.delete(:default)
```

Comment:
As an added benefit to using the injection hook `default_values` is now a getter that *always* returns a hash.